### DISCUSSION TITLE
Welcome to eeg-notebooks Discussions!

### DISCUSSION THREAD

**Discussion by JohnGriffiths** [NeuroTechX/EEG-ExPy#60] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by JohnGriffiths** [comment]
@oreHGA  @jnaulty @mhough - github discussions is activated for eeg-notebooks. 

Let's give it a spin and see how we feel.

**Reply by danielemarinazzo** [reply]
Hi everyone! I'm Daniele. Full professor of Data Analysis in the faculty of Psychology and Educational Sciences at Ghent University, this year I'm on sabbatical in Venice at San Camillo IRCCS, a research hospital. I have bought a Muse2 headband to try some simple experiments with the patients here. I'm a novice in muse and in python. I have (sort of) contributed the single frequency steady state auditory potential experiment.

**Reply by danielemarinazzo** [reply]
And here comes my first question:  are the "duration" and "trials" arguments connected? i.e. if I give a number of trials, and a certain interval between them, that would define the duration, right? I see that "duration" is defined in connection with the eeg, so it makes sense to have it as a separate thing, still I was wondering who "wins" when the number of trials would involve a longer duration than the one specified. I was now testing the SSAEP experiment, and I am only asked the duration at the CLI. Does this mean that the experiment will run until the duration reaches that number, even if there are some more trials? and that the recording would on the other hand keep going if I had chosen only a small number of trials, whose total duration is much shorter than the number I put? Maybe is too complicated to write about this in the CLI, and even more to have a check computing the minimum required time to run the experiment with the number of trials and interstimulus interval in the code and matching it with the chosen duration. Maybe in the docs?

**Reply by JohnGriffiths** [reply]
Hi @danielemarinazzo 

These are excellent questions and exactly something worth having some discussion on. 

At the moment there is no policy or intended behaviour on the primacy of trials vs duration arguments, and indeed some potential for errors if bad combinations of them are given. 

I think the best option would be for all experiments to accept EITHER n trials OR duration as input arguments, and the function then calculates the non-specified one and informs the user of the results with a printed message. 

And yes, this should be described clearly in the docs.

Can anyone think of any situation in the available experiments where this wouldn't work?

A related, third option, used by the aMMN experiment, is to load in a fixed stimulus order list. That should also be straightforward to accommodate. 

Practically,  if we wanted to do this, we would need to add a few extra lines at the start of each experiment calculating duration or n trials. 

This actually does lead to another worthwhile discussion point: we should really have an experiment class that includes all universal features, and is inherited and modified for each individual experiment. I might have a stab at that this weekend.

**Comment by oreHGA** [comment]
Hi everyone! I'm [Ore](https://oreogundipe.dev/whoami). A software engineer with a focus in data intelligence & I'm interested in understanding how the brain works. I found this project when I was scouring the web for existing work on recording and analyzing eeg recordings. It was pretty exciting collaborating with y'all in eeg-notebooks sprint at BrainHack Ontario.

At the moment, I'm familiarizing myself with the existing experiments (running them on my Notion2). I'm looking to learn how to create custom eeg experiments that can measure lived experiences (e.g being in a "distracted state" / how the brain responds to stimulations)

**Reply by JohnGriffiths** [reply]
@ErikBjare maybe he's already connected with you, but I told @oreHGA that your project has a lot of overlap with these ideas about distraction detection.
